content/Journey/_index.md

```diff
@@ -0,0 +1,98 @@
+
+- Worst: Code reviews are done solo and comments sent back to the developer. This adds the most drag.
+- Less bad: The reviewer and author meet to review and correct issues. There is still a wait time for that meeting, but it is more efficient.
+- Best: [Pair programming](https://martinfowler.com/bliki/PairProgramming.html) builds code review into the flow.
+
+### Tests are deferred or skipped
+
+We must treat tests as first-class citizens. They are more important than the code we are testing. They cannot be an afterthought and we need to make sure we are using them to increase our confidence rather than meeting some arbitrary ["coverage metric"](https://dojoconsortium.org/metrics/code-coverage/). We don't need all of the legacy code to be 100% tested, or even 1% tested to start CI. We need to commit to "we will never push untested changes." Timelines are irrelevant if we deliver things on time that don't work.
+
+### The team lacks knowledge on how to write tests for CI
+
+Dig into testing. Learn effective test patterns. Many people start out testing implementation and then struggle to keep tets current as implementation changes. Avoid implementation testing. Test behaviors. If we cannot refactor code without changing the test then we are testing incorrectly. Focus on BDD to define tests and TDD to implement tests. People often push back on TDD, but that is most commonly because they never learned how to do it correctly. BDD was created to help. Lean in.
+
+There are layers of tests in a properly architected suite of CI tests. The [*xUnit Test Patterns*](http://xunitpatterns.com/) book is a great place to start.
+
+Teams are not good at testing initally for the same reason they aren't good at any other skill on day 1. It takes time. Start now.
+
+### Individual tasks are too big
+
+CI means we are integrating partially completed features continuously. The TDD helps us learn how to decompose tasks into very small, releasable changes that are do not break existing behaviors. However, the initial habit is generally to not interage changes until they are "complete". This results is large change-sets that are more difficult to code review. This makes code review take more time, require more re-work, and reduces the ability of the reviewer to spot problems. By focusing as a team to decompose tasks to hours of work rather than days, we have improved clarity, smaller change-sets, and higher quality.
```

Comment:
```suggestion
CI means we are integrating partially completed features continuously. The TDD helps us learn how to decompose tasks into very small, releasable changes that are do not break existing behaviors. However, the initial habit is generally to not integrate changes until they are "complete". This results is large change-sets that are more difficult to code review. This makes code review take more time, require more re-work, and reduces the ability of the reviewer to spot problems. By focusing as a team to decompose tasks to hours of work rather than days, we have improved clarity, smaller change-sets, and higher quality.
```